esprima.js

```diff
@@ -4023,18 +4023,20 @@
         if (match('=')) {
             lex();
             init = isolateCoverGrammar(parseAssignmentExpression);
-        } else if (id.type !== Syntax.Identifier) {
-            expect('=');
+        } else {
+            if (!(options && options.inFor) && (id.type !== Syntax.Identifier)) {
+                expect('=');
+            }
         }
 
         return node.finishVariableDeclarator(id, init);
     }
 
-    function parseVariableDeclarationList() {
+    function parseVariableDeclarationList(options) {
```

Comment:
These "options" should not be optional. Pass them in explicitly at every call site.